pipeline_stub.py

```
@@ -0,0 +1,51 @@
+"""
+Minimal pipeline stub for CI testing.
+Provides basic functionality without heavy ML dependencies.
+"""
+
+import re
+from typing import Dict
```

Ruff (0.11.9)

`typing.Dict` is deprecated, use `dict` instead

(UP035)